tests/integ/sagemaker/jumpstart/conftest.py

```diff
@@ -65,12 +66,31 @@ def _teardown():
         endpoint_info["Endpoint"]["EndpointConfigName"]
         for endpoint_info in search_endpoints_result["Results"]
     ]
-    model_names = [
-        sagemaker_client.describe_endpoint_config(EndpointConfigName=endpoint_config_name)[
-            "ProductionVariants"
-        ][0]["ModelName"]
-        for endpoint_config_name in endpoint_config_names
-    ]
+    model_names = list(
+        filter(
+            lambda elt: elt is not None,
+            [
+                sagemaker_client.describe_endpoint_config(EndpointConfigName=endpoint_config_name)[
+                    "ProductionVariants"
+                ][0].get("ModelName")
+                for endpoint_config_name in endpoint_config_names
+            ],
+        )
+    )
+
+    inference_component_names = []
+    for endpoint_name in endpoint_names:
+        for inference_component in sagemaker_client.list_inference_components(
+            EndpointNameEquals=endpoint_name
+        ).get("InferenceComponents", []):
+            inference_component_names.append(inference_component["InferenceComponentName"])
+
+    # delete inference components for test-suite-tagged endpoints
+    for inference_component_name in inference_component_names:
+        sagemaker_client.delete_inference_component(InferenceComponentName=inference_component_name)
+
+    # wait for sagemaker to detect IC has been deleted
+    time.sleep(60 * 5)
```

Comment:
wait 5 minutes?? Shouldn't we poll instead?

Reply:
theoretically, yes. but this is a relatively short test, so adding 5 minutes will not affect overall integ test execution time.